ReSwift/CoreTypes/ActionCreator.swift

```diff
@@ -10,9 +10,18 @@ import Foundation
 
 public protocol ActionCreator {
     
-    associatedtype State
+    associatedtype State: StateType
+    /**
+     Generates one or more new actions from the current state.  May be implemented asynchronously or synchronously
+     - parameter actionBlock This may be called more than once with different actions
+     */
+    func createActions(currentState state: State, actionBlock:(Action -> Void))
```

Comment:
Also why is Action a parameter of the actionBlock? Sure it should be () -> Action? as the ActionCreator can choose to return an Action to be performed immediately, or can defer and create actions at a later date asynchronously.

Can we not then redefine the actionBlock as (State -> Action?) and get rid of currentState?